pandas/tests/reshape/test_merge.py

```diff
@@ -1350,6 +1359,138 @@ def f():
             household.join(log_return, how='outer')
         pytest.raises(NotImplementedError, f)
 
```

Comment:
can you make this a separe test class. Try to use fixtures for these df's as well (and/or parametrize)

Reply:
Done. See ``TestMergeColumnAndIndex``